src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            contactsAddrs.contacts[i].MailingStreet = contactsAddrs.addrs[i].MailingStreet__c;
+            contactsAddrs.contacts[i].MailingCity = contactsAddrs.addrs[i].MailingCity__c;
+            contactsAddrs.contacts[i].MailingState = contactsAddrs.addrs[i].MailingState__c;
+            contactsAddrs.contacts[i].MailingPostalCode = contactsAddrs.addrs[i].MailingPostalCode__c;
+            contactsAddrs.contacts[i].MailingCountry = contactsAddrs.addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        // verify that we don't have new Addresses
+        List<Address__c> ListAddr = [select Id from Address__c];
+        system.assertEquals(2, ListAddr.size());
+    }
+
+    /*********************************************************************************************************
+    @description Create N new duplicate Addresses from Contact, with non-default existing  
+    verify:
+        duplicates not created
+        found match becomes Default
+    **********************************************************************************************************/            
+    @isTest
+    public static void newDupeAddrNonDefault() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // create additional non-default addresses
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'New Street' + i;
+            addrs[i].MailingCity__c = 'New City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;        
+ 
+        // create additional addresses thru the Contact.
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = addrs[i].MailingStreet__c;
+            contactsAddrs.contacts[i].MailingCity = addrs[i].MailingCity__c;
+            contactsAddrs.contacts[i].MailingState = addrs[i].MailingState__c;
+            contactsAddrs.contacts[i].MailingPostalCode = addrs[i].MailingPostalCode__c;
+            contactsAddrs.contacts[i].MailingCountry = addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.